lib/ConstExtract/ConstExtract.cpp

```diff
@@ -119,28 +120,68 @@ parseProtocolListFromFile(StringRef protocolListFilePath,
   return true;
 }
 
+static std::string extractLiteralOutput(Expr *expr) {
+  std::string LiteralOutput;
+  llvm::raw_string_ostream OutputStream(LiteralOutput);
+  expr->printConstExprValue(&OutputStream, nullptr);
+
+  return LiteralOutput;
+}
+
 static std::shared_ptr<CompileTimeValue>
 extractPropertyInitializationValue(VarDecl *propertyDecl) {
   auto binding = propertyDecl->getParentPatternBinding();
   if (binding) {
     auto originalInit = binding->getOriginalInit(0);
     if (originalInit) {
-      std::string LiteralOutput;
-      llvm::raw_string_ostream OutputStream(LiteralOutput);
-      originalInit->printConstExprValue(&OutputStream, nullptr);
-      if (!LiteralOutput.empty())
+      auto LiteralOutput = extractLiteralOutput(originalInit);
+      if (!LiteralOutput.empty()) {
         return std::make_shared<RawLiteralValue>(LiteralOutput);
+      }
+
+      if (auto callExpr = dyn_cast<CallExpr>(originalInit)) {
+        std::vector<FunctionParameter> parameters;
+
+        const auto args = callExpr->getArgs();
+        for (auto arg : *args) {
+          auto label = arg.getLabel().str().str();
+          auto expr = arg.getExpr();
+
+          switch (expr->getKind()) {
+          case ExprKind::DefaultArgument: {
+            auto defaultArgument = dyn_cast<DefaultArgumentExpr>(expr);
```

Comment:
Is it at all desirable to use `dynamic_cast` here, since `dyn_cast` wraps an implicit redundant check?

Reply:
Great catch! I'm not sure I'm familiar with `dynamic_cast`, but I think we can directly use `cast` here since we are checking the kind.

Reply:
I don't want to prematurely optimize, I was just curious since I learned that `dyn_cast` is LLVM API. If keeping it as-is is less hairy, it's probably fine to leave.